lib/blocks.php

```diff
@@ -372,3 +372,30 @@ function gutenberg_register_legacy_social_link_blocks() {
 }
 
 add_action( 'init', 'gutenberg_register_legacy_social_link_blocks' );
+
+/**
+ * Migrate the legacy `sync_status` meta key to the new `wp_pattern_sync_status` meta key.
+ *
+ * @since 16.1.1
+ *
+ * @param mixed  $value     The value to return, either a single metadata value or an array of values depending on the value of $single.
+ * @param int    $object_id ID of the object metadata is for.
+ * @param string $meta_key  Metadata key.
+ * @param bool   $single    Whether to return only the first value of the specified $meta_key.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/52232
+ */
+function gutenberg_legacy_wp_block_post_meta( $value, $object_id, $meta_key, $single ) {
+	if ( 'wp_pattern_sync_status' !== $meta_key ) {
```

Comment:
Just to confirm, for posts already created with `'sync_status'`, they are assigned the new metadata key in `gutenberg_wp_block_register_post_meta`.

Reply:
it is a bit weird how metakeys works - they are assigned whether they were explicitly set or not, so if sync_status was actually assigned previously then its value is mapped to 'wp_pattern_sync_status' in `gutenberg_wp_block_register_post_meta`

Reply:
Yep, thanks for the explainer!